modules/formulize/admin/formindex.php

```diff
@@ -112,7 +111,7 @@ function patch40() {
         }
     }
 
-    if (!$needsPatch AND !file_exists(XOOPS_ROOT_PATH.'/modules/formulize/custom_code') AND (!isset($_GET['op']) OR ($_GET['op'] != 'patch40' AND $_GET['op'] != 'patchDB'))) {
+    if (!$needsPatch AND primaryRelationshipExists() AND !file_exists(XOOPS_ROOT_PATH.'/modules/formulize/custom_code') AND (!isset($_GET['op']) OR ($_GET['op'] != 'patch40' AND $_GET['op'] != 'patchDB'))) {
```

Comment:
We need to make the custom code detection smarter... if code folder has only one file, and there are code features to convert... needs to be its own function :(